plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -271,15 +271,22 @@ fu_thunderbolt_device_probe(FuDevice *device, GError **error)
 	if (!FU_DEVICE_CLASS(fu_thunderbolt_device_parent_class)->probe(device, error))
 		return FALSE;
 
+	self->devpath = g_strdup(fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device)));
+
 	/* device */
 	if (g_strcmp0(tmp, "thunderbolt_device") == 0) {
+		g_autofree gchar *parent_name =
+		    fu_udev_device_get_parent_name(FU_UDEV_DEVICE(self));
+		/* determine if host controller or not */
+		if (parent_name != NULL && g_str_has_prefix(parent_name, "domain"))
+			self->device_type = FU_THUNDERBOLT_DEVICE_TYPE_HOST_CONTROLLER;
```

Comment:
doesn't this exact same code already run in `fu_thunderbolt_device_setup_controller`?

Reply:
I am moving it on this patch earlier so I can stop failing on device_setup if we cannot read the version